vespa-hadoop/src/main/java/com/yahoo/vespa/hadoop/pig/VespaDocumentOperation.java

```diff
@@ -182,12 +190,47 @@ public static String create(Operation op, String docId, Map<String, Object> fiel
     @SuppressWarnings("unchecked")
     private static void writeField(String name, Object value, Byte type, JsonGenerator g, Properties properties, Schema schema, Operation op, int depth) throws IOException {
         if (shouldWriteField(name, properties, depth)) {
-            g.writeFieldName(name);
-            if (shouldWritePartialUpdate(op, depth)) {
-                writePartialUpdate(value, type, g, name, properties, schema, op, depth);
-            } else {
-                writeValue(value, type, g, name, properties, schema, op, depth);
+            if (isPartialOperation(REMOVE_BAG_AS_MAP_FIELDS, name, properties, g, PARTIAL_UPDATE_REMOVE, false) ||
+                    isPartialOperation(ADD_BAG_AS_MAP_FIELDS, name, properties, g, PARTIAL_UPDATE_ASSIGN, false)){
```

Comment:
can we check it's a partial operation first, then generate the field name according to its data type?
currently, u specifically check it's a `remove/add _bag_as_map_fields`, then transform the bag to field name directly.
It'll be more general to depend on data type than on field name.

Reply:
per discussion, cannot merge writeMapPartialUpdate into writePartialUpdate method because we have to generate special key name of map partial update fields.